packages/rum-core/src/domain/assembly.spec.ts

```diff
@@ -575,6 +575,67 @@ describe('rum assembly', () => {
       })
     })
   })
+
+  describe('STREAM event processing', () => {
+    it('should convert STREAM events to VIEW events', () => {
+      const { lifeCycle, serverRumEvents } = setupAssemblyTestWithDefaults({})
+
+      const streamData = {
+        id: 'stream-id-123',
+        document_version: 42,
+        time_spent: 5000000000, // 5 seconds in nanoseconds
+      }
+
+      notifyRawRumEvent(lifeCycle, {
+        rawRumEvent: createRawRumEvent(RumEventType.STREAM, {
+          stream: streamData,
+          view: { id: 'original-view-id', url: '/test' },
+        }),
+      })
+
+      expect(serverRumEvents.length).toBe(1)
+      const resultEvent = serverRumEvents[0]
+
+      expect(resultEvent.type).toBe('view')
+    })
+
+    it('should map stream properties correctly in converted VIEW event', () => {
+      const { lifeCycle, serverRumEvents } = setupAssemblyTestWithDefaults({})
+
+      const streamData = {
+        id: 'stream-id-456',
+        document_version: 25,
+        time_spent: 3000000000, // 3 seconds in nanoseconds
+      }
+
+      notifyRawRumEvent(lifeCycle, {
+        rawRumEvent: createRawRumEvent(RumEventType.STREAM, {
+          stream: streamData,
+          view: { id: 'original-view-id', url: '/test-page' },
+        }),
+      })
+
+      expect(serverRumEvents.length).toBe(1)
```

Comment:
should there be an assertion for the event having a `stream` attribute at this stage?

Reply:
Added!